### PR TITLE
Fix #3186 - create leaf/non-leaf/requires_grad/retain_grad tutorial

### LINKED ISSUE #3186: Writing a gradient tutorial, focused on leaf vs non leaf tensors.
There is no tutorial that specifically talks about requires_grad, retain_grad, and leaf tensor/ non-leaf tensors and how they interact with each other. Can I write a tutorial specifically talking about this topic? This will be useful when gradients are used in unusual places, as is the case for the deep dream algorithm. 

cc: @albanD

### PR DESCRIPTION
**Pull request by j-silv**
This was originally bundled with PR #3389, but now broken into two separate tutorials after discussing with PyTorch team.

Fixes #3186

## Description

Add a tutorial which goes over the subtleties of leaf, non-leaf, and `requires_grad` vs. `retain_grad`

Link to [preview is here](https://docs-preview.pytorch.org/pytorch/tutorials/3492/beginner/understanding_leaf_vs_nonleaf_tutorial.html) (wait for docs to build it): 

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

### CODE REVIEW

**Comment by svekars** [comment]
cc: @soulitzer

File: index.rst [hunk 1]
```diff
@@ -99,6 +99,13 @@ Welcome to PyTorch Tutorials
    :link: intermediate/pinmem_nonblock.html
    :tags: Getting-Started
 
+.. customcarditem::
+   :header: Understanding requires_grad, retain_grad, Leaf, and Non-leaf Tensors
+   :card_description: Learn the subtleties of requires_grad, retain_grad, leaf, and non-leaf tensors
+   :image: _static/img/thumbnails/cropped/understanding_leaf_vs_nonleaf.png
+   :link: beginner/understanding_leaf_vs_nonleaf_tutorial.html
+   :tags: Getting-Started
+
```
Comment (svekars) [inline index.rst:108]: make sure to add to one of the toctrees, suggesting the intro.rst

**Comment by svekars** [comment]
[Preview](https://docs-preview.pytorch.org/pytorch/tutorials/3492/beginner/understanding_leaf_vs_nonleaf_tutorial.html) looks pretty good - make sure to add into a toctree since it'll be an orphan otherwise.

@soulitzer - do you mind taking a look?

**Review by soulitzer — APPROVED** [review]
Sorry for the delay, just got back from PTO.
This looks good to me, thanks for the great work!

**Comment by j-silv** [comment]
> Looks great! Thank you so much and sorry for the delay in reviewing.

No worries! I know everyone is super busy :)

**Comment by svekars** [comment]
The only suggestion for improvement (can be a separate PR) is to have diagrams to be in mermaid format so they are supported in both dark and light mode.
